packages/pn-personafisica-webapp/src/components/DomicileBanner/DomicileBanner.tsx

```diff
@@ -1,90 +1,168 @@
+};
+
+const getOpenStatusFromSession = () => {
+  const sessionClosed = sessionStorage.getItem('domicileBannerClosed');
+  // validate data
+  if (sessionClosed === 'true' || sessionClosed === 'false') {
+    return Boolean(sessionClosed);
+  }
+  return false;
+};
+
+const getDomicileData = (
+  source: ContactSource,
+  hasSercqSend: boolean,
+  hasCourtesyAddresses: boolean,
+  dodDisabled: boolean,
+  hasAppIO?: boolean
+): DomicileBannerData | null => {
+  const sessionClosed = getOpenStatusFromSession();
+  if (!dodDisabled && source !== ContactSource.RECAPITI && !hasSercqSend && !sessionClosed) {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    return {
+      destination: ChannelType.SERCQ_SEND,
+      operation: ContactOperation.ADD,
+      severity: 'info',
+      message: 'no-sercq-send',
+      canBeClosed: true,
+      callToAction: 'no-sercq-cta',
+    };
+  } else if (
+    source !== ContactSource.RECAPITI &&
+    ((!hasSercqSend && sessionClosed) || dodDisabled) &&
+    !hasCourtesyAddresses
+  ) {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    return {
+      destination: ChannelType.EMAIL,
+      operation: ContactOperation.SCROLL,
+      severity: 'info',
+      message: 'no-courtesy-no-sercq-send',
+      canBeClosed: false,
+      callToAction: 'no-courtesy-no-sercq-send-cta',
+    };
+  } else if (
+    !dodDisabled &&
+    source !== ContactSource.RECAPITI &&
+    hasSercqSend &&
+    !hasAppIO &&
+    hasCourtesyAddresses
+  ) {
+    return {
+      destination: ChannelType.IOMSG,
+      operation: ContactOperation.SCROLL,
+      severity: 'info',
+      message: 'no-io',
+      canBeClosed: true,
+      callToAction: 'add-io',
+    };
+  } else if (!dodDisabled && hasSercqSend && !hasCourtesyAddresses) {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    return {
+      destination: ChannelType.EMAIL,
+      operation: ContactOperation.SCROLL,
+      severity: 'warning',
+      message: 'no-courtesy',
+      canBeClosed: false,
+      callToAction: source === ContactSource.RECAPITI ? undefined : 'complete-addresses',
+    };
+  }
+  return null;
+};
+
+const DomicileBanner: React.FC<Props> = ({ source }) => {
+  const { t } = useTranslation(['recapiti']);
   const navigate = useNavigate();
   const dispatch = useAppDispatch();
   const open = useAppSelector((state: RootState) => state.generalInfoState.domicileBannerOpened);
-  const defaultAddresses = useAppSelector(
-    (state: RootState) => state.generalInfoState.defaultAddresses
+  const { DOD_DISABLED } = getConfiguration();
+
+  const digitalAddresses = useAppSelector(
+    (state: RootState) => state.generalInfoState.digitalAddresses
   );
 
-  const handleClose = useCallback(() => {
+  const hasSercqSend = digitalAddresses.find((addr) => addr.channelType === ChannelType.SERCQ_SEND);
+  const hasAppIO = digitalAddresses.find((addr) => addr.channelType === ChannelType.IOMSG);
```

Comment:
this condition is not completely right. we have to check that the use has appIO and that it is DISABLED